.github/workflows/check-new-releases.yaml

```diff
@@ -0,0 +1,335 @@
+          if [ $number_of_releases != 1 ]; then
+            echo "Nothing to update!"
+            exit 0
+          else
+            # set output release to true
+            echo "release-package=true" > $GITHUB_OUTPUT
+          fi
+
+      - name: List downloaded builds and compress them
+        run: ls ${{ env.OUTPUT_DIR }}; zip -r ${{ env.OUTPUT_DIR }}-${{ inputs.name }}.zip ${{ env.OUTPUT_DIR }}
+
+      - name: Store builds
+        if: ${{ steps.check-release.outputs.release-package == 'true' }}
+        uses: actions/upload-artifact@v3
+        with:
+          name: releases-${{ inputs.name }}
+          path: ${{ env.OUTPUT_DIR }}-${{ inputs.name }}.zip
+
+      - name: Generate matrix with all built versions
+        if: ${{ steps.check-release.outputs.release-package == 'true' }}
+        id: set-matrix
+        run: |
+          echo "matrix=$(ls -A1 ${{ env.OUTPUT_DIR }} | jq -R -s -c 'split("\n")[:-1]')" > $GITHUB_OUTPUT
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+      tarball_name_regex: ${{ inputs.tarball-regex }}
+      release: ${{ steps.check-release.outputs.release-package }}
+
+  release-artifact:
+    if: ${{ needs.download-builds.outputs.release == 'true' }}
+    name: Release artifact on LP and Github
+    needs: download-builds
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        version: ${{ fromJson(needs.download-builds.outputs.matrix) }}
+    steps:
+      - name: Print passed vars
+        run: |
+          echo "name: ${{ inputs.name }}"
+          echo "lp-project: ${{ inputs.lp-project }}"
+          echo "lp-repo: ${{ inputs.lp-repo }}"
+          echo "lp-branch-prefix: ${{ inputs.lp-branch-prefix }}"
+          echo "tarball-regex: ${{ inputs.tarball-regex }}"
+          echo "matrix version: ${{ matrix.version }}"
+      
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Download releases
+        uses: actions/download-artifact@v3
+        with:
+          name: releases-${{ inputs.name }}
+
+      - name: Decompress all versions and list them
+        run: unzip ${{ env.OUTPUT_DIR }}-${{ inputs.name }}.zip; ls "${{ env.OUTPUT_DIR }}";
+
+      - name: Get tarball name and software version
+        id: get-tarball-name
+        run: |
+          cd output/${{ matrix.version }}/
+          count=$( ls ${{ inputs.tarball-regex }} | wc -l )
+          if [ $count != 1 ]; then
+            echo "Tarball file not found with regex: ${{ inputs.tarball-regex }}"
+            exit 1
+          fi
+          tarball_filename=$(ls ${{ inputs.tarball-regex }})
+          echo "TARBALL_FILENAME=$tarball_filename" >> $GITHUB_ENV
+          version=$( echo "${{ matrix.version }}" | awk -F '-' '{print $NF}' )
+          echo "VERSION=$version" >> $GITHUB_ENV
+          echo "Tarball filename: ${{ env.TARBALL_FILENAME}}"
+          echo "Software version: ${{ env.VERSION }}"
```

Comment:
**suggestion** It is rather minor and it is more of a suggestion, but I'd prefer to use the GITHUB_OUTPUT over GITHUB_ENV, as it is easier to understand where the variable was defined and where it comes from